### DISCUSSION TITLE
Is there a way to use graphql-spring-boot and send graphql metrics to Apollo Studio?

### DISCUSSION THREAD

**Discussion by IvanNikolaychuk** [graphql-java-kickstart/graphql-spring-boot#773] [Q&A] [answered]
Hello, I am relatively new to GraphQL world, but I've joined the existing project that uses graphql-java-kickstart/graphql-spring-boot as a graphql server. 

We would like to add monitoring to our environment (track what queries were executed, performance, issues, exc.) and I was searching for a good graphql monitoring solution. 

My research led me to Appolo Server and studio, which seems to provide everything we need monitoring-wise. 

Is there a way of connecting to Apollo Studio and pushing metrics with existing kickstart graphql server? 

P.S. I might not use terminology correctly, please correct me if needed.

**Comment by setchy** [comment, marked as answer]
You will need to use Apollo's [federation-jvm](https://github.com/apollographql/federation-jvm) library to achieve to this.

Ref to the section under [Federated Tracing](https://github.com/apollographql/federation-jvm)  for more documentation. 

A sample/example impl can be found in that repo also. Specially you'll need to add this Bean

https://github.com/apollographql/federation-jvm/blob/1fba70d11497e2973c9e25fdd6a08159ff7099f5/spring-example/src/main/java/com/apollographql/federation/springexample/graphqljavatools/AppConfiguration.java#L43

**Reply by IvanNikolaychuk** [reply]
Thanks a lot!! I will give it a go :)
